docs/source/algorithms.rst

```diff
@@ -50,3 +50,152 @@ Using the same example file as before, we can execute a GaussFilter on our test
+
+Our first example merges MS1 spectra blockwise:
+
+.. code-block:: python
+
+    # retrieve data 
+    gh = "https://raw.githubusercontent.com/OpenMS/pyopenms-extra/master"
+    urlretrieve(gh + "/src/data/small.mzML", "test.mzML")
+
+    # load MS data and store as MSExperiment object
+    exp = MSExperiment()
+    MzMLFile().load('test.mzML', exp)
+
+    spectra = exp.getSpectra()
+    spectra_ms1 = [s for s in spectra if s.getMSLevel() == 1]
+    print(f'Number of MS1 spectra before merge are {len(spectra_ms1)}')
+
+    # merges blocks of MS1
+    merger = SpectraMerger()
+
+    merger.mergeSpectraBlockWise(exp)
+
+    spectraMerged = exp.getSpectra()
+    spectra_ms1 = [s for s in spectra if s.getMSLevel() == 1]
+    print(f'Number of MS1 spectra after merge are {len(spectra_ms1)}')
+
+
+.. code-block:: output 
+  
+    Number of MS1 spectra before merge are 183
+
+    Number of MS1 spectra after merge are 37
+
+
+Per default, the method ``mergeSpectraBlockWise`` of SpectraMerger merges 5 consectutive MS1 spectra into a block. Before the merge, we had 183 MS1 spectra. Now, we have 37 MS1 spectra left. 
+
+The modified data structure can be stored on disk:
+
+.. code-block:: python
+
+    MzMLFile().store("mergedBlockWise.mzML", exp)
+
+
+SpectraMerger includes the method ``mergeSpectraPrecursors`` which allows the merging of spectra with similar precursors. The data must have MS2 spectra for the precursor information. 
+
+.. code-block:: python 
+
+    # load MS data and store as MSExperiment object
+    exp = MSExperiment()
+    MzMLFile().load('test.mzML', exp)
+
+    spectra = exp.getSpectra()
+
+    # spectra with ms_level = 2
+    spectra_ms2 = [s for s in spectra if s.getMSLevel() == 2]
+    print(f'Number of MS2 spectra before merge are {len(spectra_ms2)}')
+
+    # merge spectra with similar precursors 
+    merger = SpectraMerger()
+    merger.mergeSpectraPrecursors(exp)
+
+    spectraMerged = exp.getSpectra()
+    spectra_ms2 = [s for s in spectra if s.getMSLevel() == 2]
+    print(f'Number of MS2 spectra after merge are {len(spectra_ms2)}')
+
+    # store modified data 
+    # MzMLFile().store("mergedSimiPrecursors.mzML", exp)
+
+.. code-block:: output
+
+    Number of MS2 spectra before merge are 53
+
+    Number of MS2 spectra after merge are 53
+
+We see that the number of MS2 spectra before and after the merge do not change. This means that the hierarchical clustering with single linkage of the basic LC-MS feature (here only RT and M/Z of the precursors) did not produce any clusters (blocks to merge). 
+
+
+SpectraMerger presents a method ``average`` to average experimental data over neighbouring spectra. The block of neighbouring spectra depends on the averaging type: ``gaussian`` or ``tophat``. The gaussian type checks for a weight < cutoff value, whereas tophat averages over a range (by default 5 steps left and right from each scan). Per default SpectraMerger averages MS1 spectra. 
```

Comment:
what are the steps here?

Reply:
One step is just one scan/ spectrum. So it would be like a window of 5 scans left and 5 scans right for each scan with averaging type tophat.

Reply:
ah, thanks. I think then we should just talk about ± 5 neighboring spectra. The user might get confused if she reads steps.